### PR TITLE
5127 new module xenium ranger

### PR DESCRIPTION
**Pull request by khersameesh24**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: "v3.1.0"
+    rev: "v4.0.0-alpha.8"
```
Comment (mashehu) [inline .pre-commit-config.yaml:3]:
```suggestion
    rev: "v3.1.0"
```
please keep the original version. We want to use the currently stable version of prettier.

Comment (khersameesh24) [reply .pre-commit-config.yaml:3]: Hi Matthias, I am getting a lot of issues with unstable checksums although the same tests when run on my local system run smoothly. Not sure how to fix this

File: modules/nf-core/xeniumranger/import-segmentation/main.nf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    input:
+    val(meta)
+    path(xenium_bundle)
+    val(expansion_distance)
+    path(coordinate_transform)
+    path(nuclei)
+    path(cells)
+    path(transcript_assignment)
+    path(viz_polygons)
```
Comment (matthdsm) [inline modules/nf-core/xeniumranger/import-segmentation/main.nf:15]: Sample specific data should go in the same tuple as `meta`.
Reference data should either go in one tuple with it's own `meta` or in separate channels per datatype, again each with it's own `meta`

example: https://github.com/nf-core/modules/blob/master/modules/nf-core/snapaligner/align/main.nf#L10

**Review by matthdsm — DISMISSED** [review]

File: modules/nf-core/xeniumranger/import-segmentation/main.nf (hunk 2 above)
Comment (khersameesh24) [reply modules/nf-core/xeniumranger/import-segmentation/main.nf]: Code Changed -> Tested -> Updated PR ✌🏻

File: modules/nf-core/xeniumranger/import-segmentation/main.nf [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    tuple val(meta), path(xenium_bundle)
+    val(expansion_distance)
+    path(coordinate_transform)
+    path(nuclei)
+    path(cells)
+    path(transcript_assignment)
+    path(viz_polygons)
+
+    output:
+    tuple val(meta), path("**/outs/**"), emit: outs
```
Comment (matthdsm) [inline modules/nf-core/xeniumranger/import-segmentation/main.nf:17]: What does the tool output? different file formats should also have different output channels

Comment (khersameesh24) [reply modules/nf-core/xeniumranger/import-segmentation/main.nf:17]: Hi Matthias, the output for xeniumranger modules is reupdated in the input files just like other 10x tools, multiple files are updated for every run and hence we do not have a dedicated channel for the outputs

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mashehu) [reply .pre-commit-config.yaml:3]: please revert this

**Review by mashehu — APPROVED** [review]

**Pull request merged**